core/templates/dev/head/pages/exploration_player/AnswerClassificationService.js

```diff
@@ -125,27 +127,35 @@ oppia.factory('AnswerClassificationService', [
           return deferred.promise;
         }
 
-        if (result.outcome === defaultOutcome &&
-            INTERACTION_SPECS[oldState.interaction.id]
-              .is_string_classifier_trainable &&
+        var ruleBasedOutcomeIsDefault = (result.outcome === defaultOutcome);
+        var interactionIsTrainable = INTERACTION_SPECS[
+          oldState.interaction.id].is_interaction_trainable;
+        if (ruleBasedOutcomeIsDefault && interactionIsTrainable &&
             ENABLE_ML_CLASSIFIERS) {
-          var classifyUrl = '/explorehandler/classify/' + explorationId;
-          var params = (
-            isInEditorMode ? {} : LearnerParamsService.getAllParams());
-
-          $http.post(classifyUrl, {
-            old_state: oldState.toBackendDict(),
-            params: params,
-            answer: answer
-          }).then(function(response) {
-            var result = response.data;
+          var classifierData = StateClassifierMappingService.getClassifierData(
+            stateName);
+          var algorithmId = StateClassifierMappingService.getAlgorithmId(
+            stateName);
+          if (classifierData && algorithmId) {
+            var predictionService = (
+              PredictionAlgorithmRegistryService.getPredictionService(
+                algorithmId));
+            var predictedAnswerGroupIndex;
+            if (predictionService) {
+              predictedAnswerGroupIndex = predictionService.predict(
+                classifierData, answer).answerGroupIndex;
+            } else {
+              predictedAnswerGroupIndex = result.answerGroupIndex;
+            }
             deferred.resolve({
               outcome: result.outcome,
```

Comment:
So once you get predictedAnswerGroupIndex do this (you will need to do this separately in **if** part and **else** part),
```
outcome: answerGroups[predictedAnswerGroup].outcome,
answerGroupIndex: predictedAnswerGroup,
ruleIndex: <iterate through rules to find fuzzy rules index>
classificationCategorization: FUZZY_RULES_CLASSIFICATION (or something similar)
```

Finding `ruleIndex`
```
for (rule_spec in answerGroups[predictedAnswerGroupIndex].rule_specs) {
    if (rule_spec.rule_type === RULE_TYPE_CLASSIFIER) {
        ruleIndex = classifier_rule_spec_index;
        break;
    }
}
```

Does that help understanding what I'm trying to say?